daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -396,7 +412,7 @@ private[lf] case class PartialTransaction(
   def insertCreate(
```

Comment:
The API is now mixing `Value` and `SValue`. I do not find that very neat. 
I would argue, either you do not change the type of `arg` or you change both `arg` and `key`.

Reply:
fair enough, especially if we end up killing `toValue` I’m fine with doing it outside of ptx.

Reply:
I agree it's a big ugly that some stuff is in ptx and other stuff outside. I will see if I can improve this.
The reason for being inside ptx is that this is where the `valueNormalization` flag is accessed.

Reply:
But since I now defined `ptx.normValue`  (didn't have this originally)
```
def normValue(templateId: TypeConName, svalue: SValue): Value[Value.ContractId]
```
then I should be able to call this from outside.

Reply:
@remyhaemmerle-da 
> The API is now mixing Value and SValue. I do not find that very neat.
I hope things are now more to your satisfaction :)

Reply:
I rediscovered the other reason why some of the normalization (speedy-value --> value) was done inside ptx...

In the case of the call to `endExercises` from `Speedy.scala`, the `templateId` was not readily available at the call site, but it was available inside ptx in the `ExercisesContextInfo`.

But I fixed it now to thread the `templateId` from `Compiler.scala` through `ScopeExercises` and into `KCloseExercise` where it is needed.